packages/engine/lib/provider/src/requisition.rs

```diff
@@ -0,0 +1,60 @@
+//! Internal API
+
+use crate::{tags, Requisition, TagValue, TypeTag};
+
+impl<'p> Requisition<'p, '_> {
+    /// Provide a value with the given [`TypeTag`].
+    pub fn provide<I>(&mut self, value: I::Type) -> &mut Self
+    where
+        I: TypeTag<'p>,
+    {
+        if let Some(res @ TagValue(Option::None)) =
+            self.0.tagged.downcast_mut::<tags::OptionTag<I>>()
+        {
+            res.0 = Some(value);
+        }
+        self
+    }
+
+    /// Provide a value or other type with only static lifetimes.
+    pub fn provide_value<T, F>(&mut self, f: F) -> &mut Self
+    where
+        T: 'static,
+        F: FnOnce() -> T,
+    {
+        self.provide_with::<tags::Value<T>, F>(f)
+    }
+
+    /// Provide a reference, note that the referee type must be bounded by `'static`, but may be
```

Comment:
_reference_ type?

Reply:
No, the referee is `'static`: `T: 'static`, not `&T: 'static`

Reply:
Right but it is `value: &'p T`. I think I understand what you're getting at but I'm not sure the wording is clear, in fact I've just googled `"referee" type rust` and there are basically no results where people use the word `referee` when talking about the type that is being referred to. Probably because referee means the thing doing the _referring_ another, rather than the thing _being_ referred. I.e. if I give you a referral, I am a referee.

Maybe even just ``note that the type must be bounded by `'static` `` would be clearer, albeit still not great?

Reply:
I think "reference/referee" means the same as "pointer/pointee". Never looked up the word but we surely can improve the wording then!

Reply:
"/// Provide a reference, note that `T` must be bounded by `'static`, but may be unsized." 😆

Reply:
Unfortunately it just seems to be one of those lovely discrepancies of English. I can't find any examples online where people use 'referee' for being referred _to_, even when specifically looking for contexts around references in CS. Let's just err on the side of correctness and not use it :)